v0/destinations/salesforce/transform.js

```diff
@@ -235,11 +235,33 @@ async function getSalesforceIdFromPayload(message, authorizationData) {
         leadObjectId = leadQueryResponse.data.searchRecords[0].Id;
       }
     }
+    if (leadObjectId) {
+      const convertedDetails = await axios.get(
+        `${authorizationData.instanceUrl}/services/data/v${SF_API_VERSION}/sobjects/Lead/${leadObjectId}?fields=IsConverted,ConvertedContactId,IsDeleted`,
+        {
+          headers: { Authorization: authorizationData.token }
+        }
+      );
 
-    // add a Lead Object to the response
-    salesforceMaps.push({ salesforceType: "Lead", salesforceId: leadObjectId });
-  }
+      if (convertedDetails.data.IsConverted) {
+        leadObjectId = convertedDetails.data.ConvertedContactId;
+      }
 
+      if (convertedDetails.data.IsDeleted === false)
+        salesforceMaps.push({
+          salesforceType: "Lead",
+          salesforceId: leadObjectId
+        });
+      else {
+        throw new CustomError("The lead/contact has been deleted.", 400);
```

Comment:
Rather than else we should check
```
if (convertedDetails.data.IsDeleted === true) {
{
        throw new CustomError("The lead/contact has been deleted.", 400);
}
```

Reply:
Thank you, I have updated, and made the check to be done first in line.